### PR TITLE
`AsyncConnection[AsyncSession].aclose` implementation 

### LINKED ISSUE #9698: async connection/session should have a "aclose" method that calls "close"
### Discussed in https://github.com/sqlalchemy/sqlalchemy/discussions/9697

<div type='discussions-op-text'>

<sup>Originally posted by **jalvespinto** April 25, 2023</sup>
With sync I use the code bellow to delete tables:
```
 with contextlib.closing(pg_engine.connect()) as con:
     trans = con.begin()
     for table in reversed(database.Base.metadata.sorted_tables):
         con.execute(table.delete())
     trans.commit()
```
I see that contextlib have a 'aclosing' method, but it calls the 'aclose' method from thing passed to it (asyncconnection in my case). Which async connection does not have. </div>

### PR DESCRIPTION
**Pull request by grigoriev-semyon**
### Checklist

This pull request is:

Fixes: #9698 

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [X] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Comment by grigoriev-semyon** [comment]
contextlib.aclosing() docs available here: https://docs.python.org/3/library/contextlib.html#contextlib.aclosing

**Comment by grigoriev-semyon** [comment]
About typo error: https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_password

File: README.unittests.rst [hunk 1]
```diff
@@ -268,7 +268,7 @@ intended for production use!
 
     # configure the database
     sleep 20
-    docker exec -ti mysql mysql -u root -ppassword -w -e "CREATE DATABASE test_schema CHARSET utf8mb4; GRANT ALL ON test_schema.* TO scott;"
+    docker exec -ti mysql mysql -u root -p -w -e "CREATE DATABASE test_schema CHARSET utf8mb4; GRANT ALL ON test_schema.* TO scott;"
```
Comment (CaselIT) [inline README.unittests.rst:271]: this seems unrelated, should be reverted

File: README.unittests.rst [hunk 2]
```diff
@@ -280,7 +280,7 @@ intended for production use!
 
     # configure the database
     sleep 20
-    docker exec -ti mariadb mysql -u root -ppassword -w -e "CREATE DATABASE test_schema CHARSET utf8mb4; GRANT ALL ON test_schema.* TO scott;"
+    docker exec -ti mariadb mysql -u root -p -w -e "CREATE DATABASE test_schema CHARSET utf8mb4; GRANT ALL ON test_schema.* TO scott;"
```
Comment (CaselIT) [inline README.unittests.rst:283]: here too

**Review by CaselIT — CHANGES_REQUESTED** [review]
Changes look good, just a couple of unrelated things to revert

File: README.unittests.rst (hunk 1 above)
Comment (grigoriev-semyon) [reply README.unittests.rst]: done

File: README.unittests.rst (hunk 2 above)
Comment (grigoriev-semyon) [reply README.unittests.rst]: donee

**Review by CaselIT — APPROVED** [review]
thanks, running the ci on gerrit now

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 3b00b5e1c89dacd94d5e4482938617d1a854ac21 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by CaselIT** [comment]
seems not. there are problems with the bot it seems. will retry later

**Comment by zzzeek** [comment]
this is all stupid s*** with this dumb git feature "dubious ownership of repository", I have added exceptions all over the server and they keep not taking effect

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 3b00b5e1c89dacd94d5e4482938617d1a854ac21 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Failed to create a gerrit review, git squash against branch 'main' failed

**Comment by zzzeek** [comment]
then here's the merge issue

```
[Sun Jul 16 11:44:33.794762 2023] [wsgi:error] [pid 297960:tid 297963] [remote 140.82.115.106:15317] DEBUG:publishthing.publishthing.shell:git pull https://github.com/grigoriev-semyon/sqlalchemy.git aclose --squash
From https://github.com/grigoriev-semyon/sqlalchemy
 * branch                aclose     -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.

```


f***ing git

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 3b00b5e1c89dacd94d5e4482938617d1a854ac21 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 3b00b5e1c89dacd94d5e4482938617d1a854ac21: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Comment by zzzeek** [comment]
the dragon is dead.  i have slain the dragon

**Comment by CaselIT** [comment]
So you needed to specify how to merge?

**Comment by zzzeek** [comment]
yes, git just decided they would change "git pull" to no longer work automatically unless you added additional configuration

**Comment by grigoriev-semyon** [comment]
Ahah, maybe this bot hates me?))

File: lib/sqlalchemy/ext/asyncio/engine.py [hunk 3]
```diff
@@ -476,6 +476,10 @@ async def close(self) -> None:
         """
         await greenlet_spawn(self._proxied.close)
 
+    async def aclose(self) -> None:
+        """Call the close() method of :class:`_asyncio.AsyncConnection`"""
```
Comment (sqla-tester) [inline lib/sqlalchemy/ext/asyncio/engine.py:480]: **mike bayer** (zzzeek) wrote:

needs `.. versionadded: 2.0.20 ``

also add a line why we have `aclose()` and how it differs from `close()` (answer seems to be: "this is a synonym for `close()` that is compatible with context managers such as `contextlib.closing()`

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

File: lib/sqlalchemy/ext/asyncio/session.py [hunk 4]
```diff
@@ -973,6 +973,10 @@ async def close(self) -> None:
         """
         await greenlet_spawn(self.sync_session.close)
 
+    async def aclose(self) -> None:
+        """Call the close() method of :class:`_asyncio.AsyncSession`"""
```
Comment (sqla-tester) [inline lib/sqlalchemy/ext/asyncio/session.py:977]: **mike bayer** (zzzeek) wrote:

needs `.. versionadded: 2.0.20 ``

also add a line why we have `aclose()` and how it differs from `close()` (answer seems to be: "this is a synonym for `close()` that is compatible with context managers such as `contextlib.closing()`

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Review by sqla-tester — COMMENTED** [review]
**mike bayer** (zzzeek) wrote:

looks great, needs a changelog file.  also each docstring should likely mention the rationale for this method

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

File: lib/sqlalchemy/ext/asyncio/engine.py (hunk 3 above)
Comment (grigoriev-semyon) [reply lib/sqlalchemy/ext/asyncio/engine.py]: done

File: lib/sqlalchemy/ext/asyncio/session.py (hunk 4 above)
Comment (grigoriev-semyon) [reply lib/sqlalchemy/ext/asyncio/session.py]: done

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 9dbe87324d496323f1da4a131b4c3d485a8eea1b of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 9dbe87324d496323f1da4a131b4c3d485a8eea1b added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

this needs another look if you have a moment

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

seems ok

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

looks ok.

sorry I thought I had reviewed this already

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

you and I both did!  but we missed that it didnt add the scoped_session method and added it to exclusions instead :)

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4785 has been **merged**. Congratulations! :)
